src/main/java/com/cinema/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package com.cinema.service.impl;
+
+import com.cinema.dao.UserDao;
+import com.cinema.lib.Inject;
+import com.cinema.lib.Service;
+import com.cinema.model.User;
+import com.cinema.service.UserService;
+import java.util.Optional;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        return userDao.add(user);
```

Comment:
All the business logic somehow related to initializing or setting the values to our fields must be on the service layer.
If somebody creates the object _User_ not throughout the _register_ method but using the _userService.create()_ instead, we'll not have its password hashed, which might cause some mistakes in the future while working with the object or with the security itself.
So let's move the hashing logic to this method.